test/system/030-run.bats

```diff
@@ -1212,6 +1212,26 @@ EOF
     run_podman rm -f -t0 $ctr
 }
 
+@test "podman run - custom static_dir" {
+    # regression test for #19938 to make sure the cleanup process uses the same
+    # static_dir and writes the exit code.  If not, podman-run will run into
+    # it's 20 sec timeout waiting for the exit code to be written.
+
+    skip_if_remote "CONTAINERS_CONF_OVERRIDE redirect does not work on remote"
+    containersconf=$PODMAN_TMPDIR/containers.conf
+    static_dir=$PODMAN_TMPDIR/static_dir
+cat >$containersconf <<EOF
+[containers]
+static_dir="$static_dir"
+EOF
+    ctr=$(random_string)
+    CONTAINERS_CONF_OVERRIDE=$containersconf PODMAN_TIMEOUT=20 run_podman run $IMAGE true
+    # Since the container has been run with custom static_dir (where the libpod
+    # DB is stored), the default podman should not see it.
+    run_podman 1 container exists $ctr
```

Comment:
So.... there's something broken going on.

First and foremost: ceci n'est pas une test, because you don't run with `--name $ctr`. So of course it will never exist.

However....... fixing the test causes it to break:
```diff
diff --git a/test/system/030-run.bats b/test/system/030-run.bats
index f1fd2e2a9..480a4a9db 100644
--- a/test/system/030-run.bats
+++ b/test/system/030-run.bats
@@ -1225,7 +1225,7 @@ cat >$containersconf <<EOF
 static_dir="$static_dir"
 EOF
     ctr=$(random_string)
-    CONTAINERS_CONF_OVERRIDE=$containersconf PODMAN_TIMEOUT=20 run_podman run $IMAGE true
+    CONTAINERS_CONF_OVERRIDE=$containersconf PODMAN_TIMEOUT=20 run_podman run --name $ctr $IMAGE true
     # Since the container has been run with custom static_dir (where the libpod
     # DB is stored), the default podman should not see it.
     run_podman 1 container exists $ctr
```
```console
   [08:25:50.985490553] $ /home/esm/src/atomic/2018-02.podman/libpod/bin/podman container exists pYDUQMq0zR
   #/vvvvvvvvvvvvvvvvvvvvvvvvvvvvvvvvvvvvvvvvvvvvvvvvvv
   #| FAIL: exit code is 0; expected 1
   #\^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
```
You can take it from here. And when you do, please also add `cid="$output"` after the `podman run`, and do an `exists` test on that. Just to be paranoid.

Reply:
> ceci n'est pas une test,

Mince alors :laughing:

Reply:
> You can take it from here. And when you do, please also add cid="$output" after the podman run

I only get the ID with `-d` but that will not trigger the right code paths.  But I can do an inspect to get it :+1: